tests/scanner/scanner_test.py

```diff
@@ -13,49 +13,159 @@
 from google.cloud.forseti.common.util import string_formats
+from google.cloud.forseti.common.util.index_state import IndexState
 from google.cloud.forseti.scanner import scanner
-from google.cloud.forseti.scanner.audit import iam_rules_engine as ire
-from google.cloud.forseti.scanner.scanners import iam_rules_scanner as irs
+from google.cloud.forseti.services import db
+from google.cloud.forseti.services.scanner import dao as scanner_dao
+from tests.services.util.db import create_test_engine_with_file
+from tests.unittest_utils import ForsetiTestCase
+from tests.services.scanner import scanner_dao_test
+
 
+Session = sessionmaker()
 
-class ScannerRunnerTest(ForsetiTestCase):
 
-    FAKE_global_configs = {
-        'db_host': 'foo_host',
-        'db_user': 'foo_user',
-        'db_name': 'foo_db',
-        'email_recipient': 'foo_email_recipient'
-    }
+FAKE_GLOBAL_CONFIGS = {
+    'db_host': 'foo_host',
+    'db_user': 'foo_user',
+    'db_name': 'foo_db',
+    'email_recipient': 'foo_email_recipient'
+}
 
-    FAKE_SCANNER_CONFIGS = {'output_path': 'foo_output_path'}
+NO_SCANNERS = {'scanners': [
+    {'name': 'bigquery', 'enabled': False},
+    {'name': 'bucket_acl', 'enabled': False},
+    {'name': 'cloudsql_acl', 'enabled': False},
+    {'name': 'iam_policy', 'enabled': False}
+]}
+
+TWO_SCANNERS = {'scanners': [
+    {'name': 'bigquery', 'enabled': False},
+    {'name': 'bucket_acl', 'enabled': True},
+    {'name': 'cloudsql_acl', 'enabled': False},
+    {'name': 'iam_policy', 'enabled': True}
+]}
+
+class ScannerRunnerTest(scanner_dao_test.DatabaseTest):
 
     def setUp(self):
-        fake_utcnow = datetime(
-            year=1900, month=1, day=1, hour=0, minute=0, second=0,
-            microsecond=0)
-        self.fake_utcnow = fake_utcnow
-        self.fake_utcnow_str = self.fake_utcnow.strftime(
-            string_formats.TIMESTAMP_TIMEZONE_FILES)
-        self.fake_timestamp = '123456'
-        self.scanner = scanner
-        self.scanner.LOGGER = mock.MagicMock()
-        self.scanner.FLAGS = mock.MagicMock()
-        self.scanner.FLAGS.rules = 'fake/path/to/rules.yaml'
-        self.scanner.FLAGS.list_engines = None
-        self.ire = ire
-        self.irs = irs
-
-        self.fake_main_argv = []
-        self.fake_org_policies = fake_iam_policies.FAKE_ORG_IAM_POLICY_MAP
-        self.fake_project_policies = \
-            fake_iam_policies.FAKE_PROJECT_IAM_POLICY_MAP
+        """Setup method."""
+        super(ScannerRunnerTest, self).setUp()
+
+    def tearDown(self):
+        """Tear down method."""
+        super(ScannerRunnerTest, self).tearDown()
+
+    @mock.patch(
+        'google.cloud.forseti.scanner.scanners.iam_rules_scanner.iam_rules_engine', autospec=True)
+    @mock.patch(
+        'google.cloud.forseti.scanner.scanners.bucket_rules_scanner.buckets_rules_engine', autospec=True)
+    @mock.patch(
+        'google.cloud.forseti.services.server.ServiceConfig', autospec=True)
+    def test_with_runnable_scanners(
```

Comment:
This test name does not quite reflect what is being tested.  The name might be more clear as `test_scanner_index_id_can_be_correctly_initialized()`

Reply:
Done.